src/workload.rs

```diff
@@ -191,20 +216,69 @@ pub fn byte_to_ip(b: &bytes::Bytes) -> Result<IpAddr, WorkloadError> {
     }
 }
 
+impl From<&PortList> for HashMap<u16, u16> {
+    fn from(value: &PortList) -> Self {
+        value
+            .ports
+            .iter()
+            .map(|p| (p.service_port as u16, p.target_port as u16))
+            .collect()
+    }
+}
+
 impl TryFrom<&XdsWorkload> for Workload {
     type Error = WorkloadError;
     fn try_from(resource: &XdsWorkload) -> Result<Self, Self::Error> {
         let resource: XdsWorkload = resource.to_owned();
 
-        let mut waypoint_addresses: Vec<IpAddr> = Vec::new();
-        for addr in &resource.waypoint_addresses {
-            waypoint_addresses.push(byte_to_ip(addr)?)
-        }
+        let wp: GatewayAddress = match &resource.waypoint {
+            Some(w) => GatewayAddress {
+                address: match &w.address {
+                    Some(a) => match a {
+                        xds::istio::workload::gateway_address::Address::Ip(ip) => {
+                            Some(gatewayaddress::Address::IP(byte_to_ip(ip)?))
+                        }
+                        xds::istio::workload::gateway_address::Address::Hostname(hn) => {
+                            Some(gatewayaddress::Address::Hostname(hn.clone()))
+                        }
+                    },
+                    None => None,
```

Comment:
This should be an error

Reply:
implemented in https://github.com/istio/ztunnel/pull/486/commits/38d0a5394347413e198ce5afd6e82ce1f07c1913